### PR TITLE
Change Dockerfile base image to Debian

### PR DESCRIPTION
**Pull request by dav3r**
# 

## 🗣 Description
This PR changes the base Docker image for `cyhy-core` from `python:2-stretch` to `debian:stretch-slim`.

I also cleaned up the README a bit, since it had some outdated Docker image name references.

## 💭 Motivation and Context
This change was done because the Docker image was no longer building correctly on `python:2-stretch`.  The path of least resistance was to move to creating the image on Debian, similar to how we install `cyhy-core` with [`ansible-role-cyhy-core`](https://github.com/cisagov/ansible-role-cyhy-core).

## 🧪 Testing
The Docker image builds successfully- what could possibly go wrong?

## 🚥 Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by jsf9k — APPROVED** [review]
Thanks for cleaning this up!

File: README.md [hunk 1]
```diff
@@ -51,12 +51,12 @@ Whenever an aliased CyHy command is executed, it will use the current working di
 
 ### Advanced configuration
 
-By default, the container will look for your CyHy configurations in `/etc/cyhy`.  This location can be changed by setting the `CYHY_CONF_DIR` environment variable to point to your CyHy configuration directory.  The commands will also attempt to run using the `cisa/cyhy-core` image.  A different image can be used by setting the `CYHY_CORE_IMAGE` environment variable to the image name.
+By default, the container will look for your CyHy configurations in `/etc/cyhy`.  This location can be changed by setting the `CYHY_CONF_DIR` environment variable to point to your CyHy configuration directory.  The commands will also attempt to run using the `cisagov/cyhy-core` image.  A different image can be used by setting the `CYHY_CORE_IMAGE` environment variable to the image name.
 
 Example:
 ```
```
Comment (mcdonnnj) [inline README.md]: I think this should have `bash` formatting.

**Review by mcdonnnj — APPROVED** [review]
Thanks for getting this straightened out! 💪

File: README.md (hunk 1 above)
Comment (dav3r) [reply README.md]: Thanks - nice catch!  Added in https://github.com/cisagov/cyhy-core/pull/50/commits/0e7619fc563f67eb9d933bb0e358983165428a29.

**Review by hillaryj — APPROVED** [review]
⚾

**Pull request merged**